security/security.md

```diff
@@ -1,18 +1,194 @@
+
+The best protection against this kind of attack is to use a truly strong password. The strongest passwords are relatively long and randomly generated. There are several services and programs called password managers that you can use to generate and save passwords. Using a password manager makes having strong passwords much easier. For most of these programs, you only need to know a single password to open the password manager itself. They also usually encrypt their password databases so others cannot access them; this is much more secure than storing your passwords in a notebook or a Word document, for example. WordPress also will generate a strong random password for you, but you have to store the password somewhere if you want to be able to log into the WordPress admin dashboard later.
+
+There additional steps that can be taken to protect WordPress against multiple login attempts. These steps largely boil down to adding an additional layer of authentication and to limiting logins to the WordPress admin dashboard. Additional layers of authentication and limiting logins can be implemented at the server level or through WordPress plugins.
+
+Some of the following recommendations have been borrowed from [the WordPress Codex's guide on Brute Force Attacks](https://codex.wordpress.org/Hardening_WordPress).
+
+### Notes About Usernames
+Some WordPress security guides recommend using unique usernames for WordPress administrator accounts. While well-intentioned, WordPress's REST API allows anyone to view many of the users for your WordPress website. You can see this for yourself by putting /wp-json/wp/v2/users on the end of your WordPress website's URL. The output will be formatted for programs to use, but it should still be somewhat readable.
+
 ### Server Level
 
+#### Password Protect wp-login.php
+Adding a second password to wp-login.php requires all users to provide a password before being able to login into WordPress. This protects WordPress by adding a second username and password brute force attackers must guess before they can start guessing WordPress login information. Password protecting wp-login.php requires setting up a .htpasswd file. Your hosting provider may have tools that do this for you, especially if they use popular web panel software like cPanel or Plesk. The .htpasswd can be setup manually if you have access to the files on your hosting account. There are several tools, [such as this one](http://www.htaccesstools.com/htpasswd-generator/), on the Internet that will help you with generating a .htpasswd file. After you generate the file, you can put it in the same folder as your website or in a folder outside of your website's public folder. Once you have created the .htpasswd file, you have to tell the web server where to find the file. The configuration will be different depending on the web server you are using. Basic configuration examples for Apache and nginx, two of the most popular web servers, can be found below.
```

Comment:
> Password protecting wp-login.php requires setting up a .htpasswd file. Your hosting provider may have tools that do this for you, especially if they use popular web panel software like cPanel or Plesk. The .htpasswd can be setup manually if you have access to the files on your hosting account. There are several tools, [such as this one](http://www.htaccesstools.com/htpasswd-generator/), on the Internet that will help you with generating a .htpasswd file. After you generate the file, you can put it in the same folder as your website or in a folder outside of your website's public folder. Once you have created the .htpasswd file, you have to tell the web server where to find the file.

I don't know if we should go into implementation details. I'm more inclined to ask folks to contact their host. Thoughts @mikeschinkel?

Reply:
Also the `.htpasswd file` is Apache specific so I think these instructions, if we keep them, belong under that heading and not in the section intro.

Reply:
If I understand what's being recommended, I don't think I'd recommend to hosts that they add basic http-auth to installs by default.

Reply:
This is something that is new to me. As such, I was totally confused by this introductory sentence, and was only partly clear by reading the entire thing:

> _"Adding a second password to wp-login.php requires all users to provide a password before being able to login into WordPress."_  

That said, I really don't like the idea of having to set a `.htpasswd` file, nor do I like it because we use Nginx _(because Pantheon.)_

IF such a thing makes sense then I think this should be a plugin that implements this to allow the admin user to changes the global password, and then recommend that hosts use the specific plugin.  And I might be up for writing it...

Reply:
For this section, I adapted what the Codex has on brute force attacks: https://codex.wordpress.org/Brute_Force_Attacks#Password_Protect_wp-login.php

I will rewrite depending on who we are targeting. HTTP basic auth is impractical for a host to implement fleet-wide, but individual WordPress users with their own sites may find it useful.

The Codex page also seems to be of the mind that .htpasswd works for Nginx, but I admit I have not had the need to setup HTTP basic auth on Nginx before so I do not know.